utils/src/main/java/com/cloud/utils/net/NetUtils.java

```diff
@@ -268,19 +268,7 @@ public static String getDefaultEthDevice() {
             final String defDev = Script.runSimpleBashScript("/sbin/route -n get default 2> /dev/null | grep interface | awk '{print $2}'");
             return defDev;
         }
-        final String defaultRoute = Script.runSimpleBashScript("/sbin/route | grep default");
-
-        if (defaultRoute == null) {
-            return null;
-        }
-
-        final String[] defaultRouteList = defaultRoute.split("\\s+");
-
-        if (defaultRouteList.length != 8) {
-            return null;
-        }
-
-        return defaultRouteList[7];
+        return Script.runSimpleBashScript("ip route | grep default | head -1 | grep -Po '(?<=dev )(\\S+)'");
```

Comment:
@rhtyd 
when I run the command on ubuntu 18.04, I got empty result. is that normal ?

```
root@mgt118-4:~# ip route | grep default | head -1 | grep -Po '(?<=dev )(\\S+)'
root@mgt118-4:~# ip route | grep default | head -1
default via 10.11.118.254 dev eth0
```

Reply:
@weizhouapache yes, since it's a java string I've to use the double slash `\\`.
It would run this: (note the `\\S` in the string vs `\S` that it runs on bash)
`ip route | grep default | head -1 | grep -Po '(?<=dev )(\S+)'`